docs/watchlists-feeds-reports.rst

```diff
@@ -191,6 +191,94 @@ Both IOCs and reports may include a ``link`` property, which is used by the Carb
+    >> IOC_V2.create_equality(cbcsdk, None, "netconn_domain", ["localhost.local"])
+    <cbc_sdk.enterprise_edr.threat_intelligence.IOC_V2: id ad361179-d586-4c99-af3e-821224cc0fd9> @ https://<CBCInstanceURL>
+
+
+Creating a query IOC
+++++++++++++++++++++
+
+.. code-block:: python
+
+    >> IOC_V2.create_query(cbcsdk, None, "{process_hash:098f6bcd4621d373cade4e832627b4f6}")
+    <cbc_sdk.enterprise_edr.threat_intelligence.IOC_V2: id 36d68cab-4739-4aa6-afcc-2921d2e5573e> @ https://<CBCInstanceURL>
+
+
+Creating a regex IOC
+++++++++++++++++++++
+
+.. code-block:: python
+
+    >> IOC_V2.create_regex(cbcsdk, None, "process_name", r"(^/usr/.*$)|(^/bin/.*$)")
+    <cbc_sdk.enterprise_edr.threat_intelligence.IOC_V2: id 5170a04c-bbfc-4449-b939-d5fc9f55d555> @ https://<CBCInstanceURL>
+
+
+Removing and adding an IOC from a Report
+++++++++++++++++++++++++++++++++++++++++
+
+If you want to remove an IOC from a report, you will need the IOC id and the report id.
+
+.. code-block:: python
+
+    >> from cbc_sdk.enterprise_edr import Report
+    >> ioc_id = "<ioc_id>"
+    >> report = cbcsdk.select(Report).where(id="<report_id>", feed_id="<feed_id>")[0]
+    <cbc_sdk.enterprise_edr.threat_intelligence.Report: id 1e69c54e-7cc9-41b8-9d1d-3fd59a003d8a> @ https://<CBCInstanceURL>
+    >> report.remove_iocs_by_id([ioc_id])
+    >> report.update()
+    <cbc_sdk.enterprise_edr.threat_intelligence.Report: id 1e69c54e-7cc9-41b8-9d1d-3fd59a003d8b> @ https://<CBCInstanceURL> (*)
+
+
+Adding the IOC into the report works the same way:
+
+.. code-block:: python
+
+    >> from cbc_sdk.enterprise_edr import Report, IOC_V2
+    >> ioc_id = "<ioc_id>"
+    >> report = cbcsdk.select(Report).where(id="<report_id>", feed_id="<feed_id>")[0]
+    <cbc_sdk.enterprise_edr.threat_intelligence.Report: id 1e69c54e-7cc9-41b8-9d1d-3fd59a003d8a> @ https://<CBCInstanceURL>
+    >> ioc = IOC_V2.create_regex(cbcsdk, None, "process_name", r"(^/usr/.*$)|(^/bin/.*$)")
+    >> report.append_iocs([ioc])
+    >> report.update()
+    <cbc_sdk.enterprise_edr.threat_intelligence.Report: id 1e69c54e-7cc9-41b8-9d1d-3fd59a003d8b> @ https://<CBCInstanceURL> (*)
+
+
+.. note::
+    Calling the `Report.save()` method after the insertion or removal of IOC does not update the report
+    and it's likely to result in a bad call to the API.
+
+
+If the report is in a watchlist instead of a feed then you have to get the appropriate watchlist and iterate over the reports.
+
+.. code-block:: python
+
+    >> from cbc_sdk.enterprise_edr import Watchlist, Report, IOC_V2
+    >> ioc_id = "<ioc_id>"
+    >> report_id = "<report_id>"
+    >> watchlist = cbcsdk.select(Watchlist, "<watchlist_id>")
+    <cbc_sdk.enterprise_edr.threat_intelligence.Watchlist: id <watchlist_id>> @ https://<CBCInstanceURL>
+    >> ioc = IOC_V2.create_regex(cbcsdk, None, "process_name", r"(^/usr/.*$)|(^/bin/.*$)")
+    >> reports = watchlist.reports
+    >> report = [report_ for report_ in reports if report_.id == report_id][0]
```

Comment:
Hmm. This seems odd that we don't have a way to fetch the report directly for a watchlist as the API is available without the watchlist_id. I'd expect something like `cb.select(Report, "id", from_watchlist=True)`

Reply:
Any thoughts on creating a story to implement this behavior?

Reply:
Yes, I also think it's odd that this is not being supported however, you need to provide the Watchlist ID as well as the Report ID to select it so more like the current behavior with the `feed_id`.
```python
cb.select(Report, "id", watchlist_id="...")
```

Reply:
Why do you need the Watchlist id? or is that because of the CBC SDK side?

Reply:
If we don't have an `watchlist_id` or `feed_id` how do we know where to look for that report?

Reply:
Actually just saw the `from_watchlist` which is set to a boolean when the real value should be a string. I will have to check the logic more thoughtfully to create a proper solution for this.